packages/valory/customs/uniswap_pools_search/uniswap_pools_search.py

```diff
@@ -417,6 +419,255 @@ def calculate_metrics(current_pool: Dict[str, Any], coingecko_api_key: str, coin
+
+        # logging.info full response for debugging
+        response_json = response.json()
+        logging.info("Full Response:")
+        logging.info(json.dumps(response_json, indent=2))
+
+        if response.status_code == 200:
+            # Check for GraphQL errors
+            if "errors" in response_json:
+                logging.info("GraphQL Errors:")
+                logging.info(json.dumps(response_json["errors"], indent=2))
+                return None
+
+            # Check for valid pool data
+            data = response_json.get("data", {})
+            pool = data.get("pool")
+
+            if pool is None:
+                logging.info("No pool data found for the given ID")
+                return None
+
+            return pool
+        else:
+            logging.info(f"HTTP Error: {response.status_code}")
+            logging.info(f"Response Text: {response.text}")
+            return None
+
+    except requests.RequestException as e:
+        logging.error(f"Request Exception: {e}")
+        return None
+    except json.JSONDecodeError as e:
+        logging.error(f"JSON Decode Error: {e}")
+        return None
+    except Exception as e:
+        logging.error(f"Unexpected Error: {e}")
+        return None
+
+def fetch_24_hour_volume(pool_id: str) -> List[Dict[str, Union[int, float, str]]]:
+    """
+    Fetch 24-hour volume data for a specific pool from The Graph.
+
+    Args:
+        pool_id (str): The unique identifier of the pool to fetch volume data for.
+
+    Returns:
+        List[Dict[str, Union[int, float, str]]]: A list of dictionaries containing volume data,
+        or an empty list if retrieval fails.
+    """
+    timestamp_24h_ago = int(time.time()) - 86400
+    query = {
+        "query": f"""
+        {{
+          poolDayDatas(first: 1, orderBy: date, orderDirection: desc, where: {{
+            pool: "{pool_id.lower()}",
+            date_gt: {timestamp_24h_ago}
+          }}) {{
+            date
+            liquidity
+            volumeUSD
+            volumeToken0
+            volumeToken1
+          }}
+        }}
+        """
+    }
+    try:
+        response = requests.post(
+            SUBGRAPH_URL,
+            json=query,
+            headers={"Content-Type": "application/json", "Accept": "application/json"},
+        )
+
+        if response.status_code == 200:
+            data = response.json()
+            pool_day_datas = data.get("data", {}).get("poolDayDatas", [])
+
+            if not pool_day_datas:
+                logging.info("No volume data found for the pool")
+                return []
+
+            return pool_day_datas
+        else:
+            logging.info(f"Error fetching 24-hour volume: {response.status_code}")
+            logging.info(f"Response Text: {response.text}")
+            return []
+
+    except Exception as e:
+        logging.error(f"Exception in volume fetch: {e}")
+        return []
+
+def calculate_metrics(
```

Comment:
can you rename this function, as a similar one already exists

Reply:
changes made into new commit